### ISSUE TITLE
lessen usage of uids

### ISSUE THREAD

**Issue by parmentelat** [parmentelat/nbhosting#112]
right now we make sure each incoming user - mooc or classroom - gets a local uid

that looked kind of nice, but turned out to have some severe drawbacks
. for some reason (check if still true) each ui needs to be added to the docker group; 
. for that to trigger, we even need to 'kill --reload' the docker daemon more or less each time a notebook request comes in (again check if still true)
. and then, more importantly, I have established that a great number of uids was causing a serious performance hit; at some point thurst needed in the 20-25 sometimes even 30s to answer, and it came down to 8-10 when I cleaned up /etc/passwd
. so we routinely clean up; which causes the additional need to check for existence, create by need, chown... a whole lot of business

on the plus side, well, it was supposed to help isolating users data and access; it seems that it is not actually involved in any of the protection schemes that we do use 
. the website, when they all run as the webserver's uid anyway
. their container, where they have no view on the other student's data anyway

bottom line is, this is really not helping, the cons vastly outweigh the pros 

it'd be very nice to get rid of that altogether

**NOTES** 

- investigate if podman can help
- provisions should still be in place for classroom mode, where additional services like dropbox-like features could come in handy; limiting uis to classroom mode may be the way to go

**Comment by parmentelat** [comment]
the plan is to simply use a single uid in the root context, for everything, i.e.
* running the uwsgi app
* and spawning docker containers

no more chown, no more start-with-uid, as simple as it gets

foreseeable impacts are on
* uwsgi config
* nbh spawner script
* docker images and in particular a drastic simplification (if still needed at all) of start-in-dir-as-uid.sh
